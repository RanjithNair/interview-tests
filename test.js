describe('Coding Tests', () => {
  it('Increase the age attribute by 2', () => {
    let obj = [
      {
        name: 'Roger Federer',
        age: 37
      },
      {
        name: 'Rafael Nadal',
        age: 32
      },
      {
        name: 'Novak Djokovic',
        age: 30
      }
    ]
    const modifiedObj = []
    expect(obj).toMatchSnapshot()
  })

  it('Print all the keys of the objects in an array', () => {
      const obj = {
          'age': '22',
          'name': 'ABC',
          'place': 'Milwaukee',
          'designation': 'Software Developer'
      }
      const objKeys = []
      expect(objKeys).toMatchSnapshot()
  })

  it('Print all the values of the objects in an array', () => {
    const obj = {
        'age': '22',
        'name': 'ABC',
        'place': 'Milwaukee',
        'designation': 'Software Developer'
    }
    const objValues = []
    expect(objValues).toMatchSnapshot()
  })

  it('Find all the entries whose age is more than 20', () => {
    let obj = [
        {
          name: 'Mark James',
          age: 18
        },
        {
          name: 'AB Daniels',
          age: 22
        },
        {
          name: 'Thomas Jeffrey',
          age: 25
        }
      ]
    const result = []  
    expect(result).toMatchSnapshot()  
  })

  it('Find average of the numbers in an array', () => {
    const data = [29.76, 41.85, 46.5]
    const result = 0
    expect(result).toMatchSnapshot()
  })
})
