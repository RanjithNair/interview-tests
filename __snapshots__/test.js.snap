// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coding Tests Find all the entries whose age is more than 20 1`] = `
Array [
  Object {
    "age": 22,
    "name": "AB Daniels",
  },
  Object {
    "age": 25,
    "name": "Thomas Jeffrey",
  },
]
`;

exports[`Coding Tests Find all the entries whose age is more than 25 1`] = `
Array [
  Object {
    "age": 22,
    "name": "Rafael Nadal",
  },
  Object {
    "age": 25,
    "name": "Novak Djokovic",
  },
]
`;

exports[`Coding Tests Find average of the numbers in an array 1`] = `39.37`;

exports[`Coding Tests Increase the age attribute by 2 1`] = `
Array [
  Object {
    "age": 39,
    "name": "Roger Federer",
  },
  Object {
    "age": 34,
    "name": "Rafael Nadal",
  },
  Object {
    "age": 32,
    "name": "Novak Djokovic",
  },
]
`;

exports[`Coding Tests Print all the keys of the objects in an array 1`] = `
Array [
  "age",
  "name",
  "place",
  "designation",
]
`;

exports[`Coding Tests Print all the values of the objects in an array 1`] = `
Array [
  "22",
  "ABC",
  "Milwaukee",
  "Software Developer",
]
`;
